algorithms/kernel/covariance/covariance_impl.i

```diff
@@ -184,29 +185,28 @@ services::Status updateDenseCrossProductAndSums(bool isNormalized, size_t nFeatu
             algorithmFPType alpha = 1.0;
             algorithmFPType beta  = 1.0;
 
-            size_t startRow = iBlock * numRowsInBlock;
-            size_t endRow   = startRow + numRowsInBlock;
-            if (endRow > nVectors)
-            {
-                endRow = nVectors;
-            }
+            size_t _nRows    = (iBlock < (numBlocks - 1)) ? numRowsInBlock : numRowsInLastBlock;
```

Comment:
naming of variables with _* prefix is used for private fields in classes only

Reply:
fixed